integration-libs/cds/src/trending-searches/trending-searches.service.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { inject, Injectable, OnDestroy } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { CdsEndpointsService } from '../services';
+import { CdsConfig } from '../config';
+import { BaseSiteService, WindowRef } from '@spartacus/core';
+import {
+  concatMap,
+  interval,
+  last,
+  map,
+  Observable,
+  of,
+  ReplaySubject,
+  take,
+  takeWhile,
+} from 'rxjs';
+import { SearchPhrases } from './model';
+
+const TRENDING_SEARCHES_ENDPOINT_KEY = 'searchIntelligence';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class TrendingSearchesService implements OnDestroy {
+  protected httpClient = inject(HttpClient);
+  protected cdsEndpointsService = inject(CdsEndpointsService);
+  protected cdsConfig = inject(CdsConfig);
+  protected baseSiteService = inject(BaseSiteService);
+  protected winRef = inject(WindowRef);
+  protected interval: any;
+  private readonly trendingSearchesSource = new ReplaySubject<
+    SearchPhrases[]
+  >();
+  private trendingSearches$ = this.trendingSearchesSource.asObservable();
+
+  constructor() {
+    this.addTrendingSearchesListener();
+  }
+
+  private checkAvailability() {
```

Comment:
`major:` let's make this field `protected` for easier extensibility and customisation